### PR TITLE
feature/additional-metadata-2

### LINKED ISSUE #40: Enable quick access to pre-parsed metadata (CZI)
The super issue for this has more information, but it would be nice to have quick access via a simple property to grab some commonly used pieces of metadata otherwise somewhat obscured in the sea of metadata

### PR DESCRIPTION
**Pull request by BrianWhitneyAI**
### Description 
The purpose of this PR is to resolve #40 and  implement standard metadata properties released in bioio 1.5.0. This PR is the second rendition of this [PR](https://github.com/bioio-devs/bioio-czi/pull/34), repairing it from the recent pylibczirw library implementation. 

example:
```
image = bioio_czi.Reader(path, use_aicspylibczi=True)

image.standard_metadata.to_dict()
```
output:
```
{
 'Binning': '1x1',
 'Column': '4',
 'Dimensions Present': 'TCZYX',
 'Image Size C': 1,
 'Image Size T': 2,
 'Image Size X': 1848,
 'Image Size Y': 1248,
 'Image Size Z': 2,
 'Imaged By': 'sara.carlson',
 'Imaging Date': '2020-01-17',
 'Objective': '10x/0.45',
 'Pixel Size X': 0.5416666666666666,
 'Pixel Size Y': 0.5416666666666666,
 'Pixel Size Z': 2.23,
 'Position Index': 1,
 'Row': '4',
 'Timelapse': True,
 'Timelapse Interval': 60000.0,
 'Total Time Duration': 60000
}
```

Answering some previous questions (updates from @pgarrison)

1) ~This PR only covers metadata coverage for the `aicspylibczi` reader not for the new `pylibczirw` library. If we want to implement this in the future we need to come up with a different way to access these metadata points.~ This PR adds support in both of the plugin's modes: aicspylibczi and pylibczirw.

2) ~This PR inherently makes the standard metadata properties private since they're not accessible from the top level reader. Though users who directly implement `aicspylibczi` reader will be able to access them as public properties. I still don't really have a preference in this category, I slightly lean towards giving users as much access as possible.~ This PR makes the standard metadata properties available on the top-level reader.

### CODE REVIEW

**Review by kmitcham — APPROVED** [review]

**Comment by pgarrison** [comment]
With the exception of the fields that rely on subblock metadata, I think we should make this feature available in `pylibczrw` mode. And I think we can achieve that by moving all(?) of the implementation from the `aicspylibczi` reader to the top level `reader.py`. I'd be happy to take a crack at this.

**Comment by pgarrison** [comment]
I tried to use this as follows.

```python
from bioio_czi import Reader

image = Reader("./bioio_czi/tests/resources/OverViewScan.czi", use_aicspylibczi=True)

image.standard_metadata.to_dict()
```

But I got the following error.
```python
Failed to extract Binning setting: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
Traceback (most recent call last):
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 1067, in binning
    el = self._get_metadata_element("./Image/Pixels/Channel/DetectorSettings")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 992, in _get_metadata_element
    xml = XML(self.ome_metadata.to_xml())
              ^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 985, in ome_metadata
    return metadata_utils.transform_metadata_with_xslt(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/metadata.py", line 130, in transform_metadata_with_xslt
    xslt_abs_path = Path(xslt).resolve(strict=True).absolute()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/.pyenv/versions/3.11.9/lib/python3.11/pathlib.py", line 993, in resolve
    s = os.path.realpath(self, strict=strict)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen posixpath>", line 416, in realpath
  File "<frozen posixpath>", line 451, in _joinrealpath
FileNotFoundError: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
Failed to extract Imaged By: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
Traceback (most recent call last):
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 1085, in imaged_by
    el = self._get_metadata_element("./Experimenter")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 992, in _get_metadata_element
    xml = XML(self.ome_metadata.to_xml())
              ^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 985, in ome_metadata
    return metadata_utils.transform_metadata_with_xslt(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/metadata.py", line 130, in transform_metadata_with_xslt
    xslt_abs_path = Path(xslt).resolve(strict=True).absolute()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/.pyenv/versions/3.11.9/lib/python3.11/pathlib.py", line 993, in resolve
    s = os.path.realpath(self, strict=strict)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen posixpath>", line 416, in realpath
  File "<frozen posixpath>", line 451, in _joinrealpath
FileNotFoundError: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
Failed to extract Acquisition Date: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
Traceback (most recent call last):
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 1042, in imaging_date
    el = self._get_metadata_element("./Image/AcquisitionDate")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 992, in _get_metadata_element
    xml = XML(self.ome_metadata.to_xml())
              ^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 985, in ome_metadata
    return metadata_utils.transform_metadata_with_xslt(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/metadata.py", line 130, in transform_metadata_with_xslt
    xslt_abs_path = Path(xslt).resolve(strict=True).absolute()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/.pyenv/versions/3.11.9/lib/python3.11/pathlib.py", line 993, in resolve
    s = os.path.realpath(self, strict=strict)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen posixpath>", line 416, in realpath
  File "<frozen posixpath>", line 451, in _joinrealpath
FileNotFoundError: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
Failed to extract Objective: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
Traceback (most recent call last):
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 1105, in objective
    el = self._get_metadata_element("./Instrument/Objective")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 992, in _get_metadata_element
    xml = XML(self.ome_metadata.to_xml())
              ^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 985, in ome_metadata
    return metadata_utils.transform_metadata_with_xslt(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/metadata.py", line 130, in transform_metadata_with_xslt
    xslt_abs_path = Path(xslt).resolve(strict=True).absolute()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/philip.garrison/.pyenv/versions/3.11.9/lib/python3.11/pathlib.py", line 993, in resolve
    s = os.path.realpath(self, strict=strict)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen posixpath>", line 416, in realpath
  File "<frozen posixpath>", line 451, in _joinrealpath
FileNotFoundError: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
Failed to parse position index from scene name 'TR1': invalid literal for int() with base 10: 'R1'
Traceback (most recent call last):
  File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 1306, in position_index
    return int(prefix[1:])
           ^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'R1'
```

**Comment by BrianWhitneyAI** [comment]
> I tried to use this as follows.
> 
> ```python
> from bioio_czi import Reader
> 
> image = Reader("./bioio_czi/tests/resources/OverViewScan.czi", use_aicspylibczi=True)
> 
> image.standard_metadata.to_dict()
> ```
> 
> But I got the following error.
> 
> ```python
> Failed to extract Binning setting: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
> Traceback (most recent call last):
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 1067, in binning
>     el = self._get_metadata_element("./Image/Pixels/Channel/DetectorSettings")
>          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 992, in _get_metadata_element
>     xml = XML(self.ome_metadata.to_xml())
>               ^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 985, in ome_metadata
>     return metadata_utils.transform_metadata_with_xslt(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/metadata.py", line 130, in transform_metadata_with_xslt
>     xslt_abs_path = Path(xslt).resolve(strict=True).absolute()
>                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/.pyenv/versions/3.11.9/lib/python3.11/pathlib.py", line 993, in resolve
>     s = os.path.realpath(self, strict=strict)
>         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "<frozen posixpath>", line 416, in realpath
>   File "<frozen posixpath>", line 451, in _joinrealpath
> FileNotFoundError: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
> Failed to extract Imaged By: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
> Traceback (most recent call last):
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 1085, in imaged_by
>     el = self._get_metadata_element("./Experimenter")
>          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 992, in _get_metadata_element
>     xml = XML(self.ome_metadata.to_xml())
>               ^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 985, in ome_metadata
>     return metadata_utils.transform_metadata_with_xslt(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/metadata.py", line 130, in transform_metadata_with_xslt
>     xslt_abs_path = Path(xslt).resolve(strict=True).absolute()
>                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/.pyenv/versions/3.11.9/lib/python3.11/pathlib.py", line 993, in resolve
>     s = os.path.realpath(self, strict=strict)
>         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "<frozen posixpath>", line 416, in realpath
>   File "<frozen posixpath>", line 451, in _joinrealpath
> FileNotFoundError: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
> Failed to extract Acquisition Date: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
> Traceback (most recent call last):
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 1042, in imaging_date
>     el = self._get_metadata_element("./Image/AcquisitionDate")
>          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 992, in _get_metadata_element
>     xml = XML(self.ome_metadata.to_xml())
>               ^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 985, in ome_metadata
>     return metadata_utils.transform_metadata_with_xslt(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/metadata.py", line 130, in transform_metadata_with_xslt
>     xslt_abs_path = Path(xslt).resolve(strict=True).absolute()
>                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/.pyenv/versions/3.11.9/lib/python3.11/pathlib.py", line 993, in resolve
>     s = os.path.realpath(self, strict=strict)
>         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "<frozen posixpath>", line 416, in realpath
>   File "<frozen posixpath>", line 451, in _joinrealpath
> FileNotFoundError: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
> Failed to extract Objective: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
> Traceback (most recent call last):
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 1105, in objective
>     el = self._get_metadata_element("./Instrument/Objective")
>          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 992, in _get_metadata_element
>     xml = XML(self.ome_metadata.to_xml())
>               ^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 985, in ome_metadata
>     return metadata_utils.transform_metadata_with_xslt(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/metadata.py", line 130, in transform_metadata_with_xslt
>     xslt_abs_path = Path(xslt).resolve(strict=True).absolute()
>                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/philip.garrison/.pyenv/versions/3.11.9/lib/python3.11/pathlib.py", line 993, in resolve
>     s = os.path.realpath(self, strict=strict)
>         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "<frozen posixpath>", line 416, in realpath
>   File "<frozen posixpath>", line 451, in _joinrealpath
> FileNotFoundError: [Errno 2] No such file or directory: '/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/czi-to-ome-xslt/xslt'
> Failed to parse position index from scene name 'TR1': invalid literal for int() with base 10: 'R1'
> Traceback (most recent call last):
>   File "/home/philip.garrison/workspace/aics/bioio-czi/bioio_czi/aicspylibczi_reader/reader.py", line 1306, in position_index
>     return int(prefix[1:])
>            ^^^^^^^^^^^^^^^
> ValueError: invalid literal for int() with base 10: 'R1'
> ```

Hm Ill have to take a look at this. Values that cannot be parsed should be returned as None. Can you link me the file you are working with here?

**Comment by pgarrison** [comment]
I've added support and testing for the pylibczirw mode. This PR is ready to review.

The error I reported above was primarily due to an incorrect install of `czi-to-ome-xslt` on my end.

File: bioio_czi/aicspylibczi_reader/reader.py [hunk 1]
```diff
@@ -1,6 +1,3 @@
-#!/usr/bin/env python
```
Comment (kmitcham) [inline bioio_czi/aicspylibczi_reader/reader.py:1]: Is this deletion intentional?  It seems unrelated.

File: bioio_czi/aicspylibczi_reader/subblock_metadata.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    if acquisition_time_element is not None and acquisition_time_element.text:
+        try:
+            acquisition_time_from_czi_xml = acquisition_time_element.text
+            acquisition_time_as_date = datetime.datetime.strptime(
+                acquisition_time_from_czi_xml.split(".")[0], "%Y-%m-%dT%H:%M:%S"
+            )
+
+            formatted_acquisition_time = acquisition_time_as_date + datetime.timedelta(
+                microseconds=int(str(acquisition_time_from_czi_xml).split(".")[1][:-1])
```
Comment (kmitcham) [inline bioio_czi/aicspylibczi_reader/subblock_metadata.py:33]: There is an awful lot going on in this line.  It might be easier to read if you expand it with temporary variables or something.
Stringify the xml value, split it into an array, take the second element, remove the last character from that element, convert it to an int, and divide by 1000?

**Comment by BrianWhitneyAI** [comment]
@pgarrison should we add something to the table in the read me to cover the metadata extraction differences?

File: bioio_czi/aicspylibczi_reader/reader.py (hunk 1 above)
Comment (pgarrison) [reply bioio_czi/aicspylibczi_reader/reader.py:1]: Unrelated, but yes, intentional

File: bioio_czi/aicspylibczi_reader/subblock_metadata.py (hunk 2 above)
Comment (pgarrison) [reply bioio_czi/aicspylibczi_reader/subblock_metadata.py]: I replaced this whole block with a single builtin method: [30bfbe1](https://github.com/bioio-devs/bioio-czi/pull/43/commits/30bfbe1e530ede7c034c552a085edcff4fca5677)

File: bioio_czi/aicspylibczi_reader/subblock_metadata.py [hunk 3]
```diff
@@ -8,7 +11,7 @@
 log = logging.getLogger(__name__)
 
 
-def extract_acquisition_time_from_subblock_metadata(
+def _extract_acquisition_time_from_subblock_metadata(
```
[the remark this answers is not in the document]
Comment (pgarrison) [reply bioio_czi/aicspylibczi_reader/subblock_metadata.py:14]: It's a little ambiguous whether these should be private or public functions. At present, only `time_between_subblocks` is used outside of this module, but hypothetically future changes could use the other functions elsewhere. I picked the naming that matches the current state of the code, on the reasoning that we can update the naming later if it becomes used elsewhere.

File: bioio_czi/tests/test_standard_metadata.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        (
+            False,
+            "OverViewScan.czi",
+            {
+                "Binning": "Other",
+                "Column": None,
+                "Dimensions Present": "CYX",  # aicspylibczi mode has CMYX
+                "Image Size C": 1,
+                "Image Size T": None,
+                "Image Size X": 7398,  # Larger than aicspylibczi mode
```
[the remark this answers is not in the document]
Comment (pgarrison) [reply bioio_czi/tests/test_standard_metadata.py]: Resolved [c59f378](https://github.com/bioio-devs/bioio-czi/pull/43/commits/c59f3787b9032c863d112de5b5d1dfb96cf217da)

**Comment by pgarrison** [comment]
> @pgarrison should we add something to the table in the read me to cover the metadata extraction differences?

Done [c59f378](https://github.com/bioio-devs/bioio-czi/pull/43/commits/c59f3787b9032c863d112de5b5d1dfb96cf217da)

File: bioio_czi/reader.py [hunk 5]
```diff
@@ -345,11 +346,34 @@ def get_mosaic_tile_positions(self, **kwargs: int) -> List[Tuple[int, int]]:
...
     def standard_metadata(self) -> StandardMetadata:
         """
         Return the standard metadata for this reader, updating specific fields.
-        This implementation calls the base reader’s standard_metadata property
+        This implementation calls the base reader's standard_metadata property
         via super() and then assigns the new values.
         """
-        return self._implementation.standard_metadata
+        metadata = super().standard_metadata
+        metadata.binning = standard_metadata.binning(self.ome_metadata)
```
[the remark this answers is not in the document]
Comment (pgarrison) [reply bioio_czi/reader.py:375]: Added some comments [9b00ddd](https://github.com/bioio-devs/bioio-czi/pull/43/commits/9b00ddd7c6fa2a8cc37a070e531572a42e407bc3)

**Comment by pgarrison** [comment]
Substantive changes from [recent commits](https://github.com/bioio-devs/bioio-czi/pull/43/files/e2dd53c53f7f8ab67a4c6f7ad62759fc7b0ac595..b60e9e9e15dbeb890115c857f57740fcc5058109):
* Time intervals are now computed to microsecond precision instead of rounding to the nearest second, because the `StandardMetadata` schema expects a float.
* Acquisition times are parsed with the dateutil module (BSD license)

**Review by kmitcham — APPROVED** [review]

**Review by SeanLeRoy — APPROVED** [review]
Unfortunate we can't read timelapse duration from `pylibczirw` - this would impact MXS's ability to read the necessary metadata for its processing from cloud images.

**Comment by BrianWhitneyAI** [comment]
> Unfortunate we can't read timelapse duration from `pylibczirw` - this would impact MXS's ability to read the necessary metadata for its processing from cloud images.

**Comment by pgarrison** [comment]
I looked into our test cases more thoroughly and discovered some edge cases not handled by the time interval logic here, including sub-millisecond precision and differences between scenes.
